### DISCUSSION TITLE
How to add a Device in DeviceList with Advertiser mode in res/configuration.toml in device-ble (BLE)

### DISCUSSION THREAD

**Discussion by HCL-BMC** [edgexfoundry/community#88] [General]
Currently, BLE Device Service supports discovery/scanning mode only.
So, I am planning to add a feature( Advertiser mode) support in BLE Device Service.

In README file, There is an option to add device in device list.  But not mention to how to add in a advertiser mode.
Could you please clarify the what are the changes needs to be done or any documents or any pointer for this.

Thanks in Advance.

![Screenshot from 2023-04-07 17-47-53](https://user-images.githubusercontent.com/36533514/230611442-7e311de1-fb03-40d0-9ed4-0aab8132f66b.png)

**Comment by HCL-BMC** [comment]
I have added a sample device in DeviceList in res/configuration.toml file. But not sure how to add a device in advertiser mode.

Any suggestions on this ?

**Comment by lenny-goodell** [comment]
Please move to newer supported 2.x release of EdgeX where the DeviceList  is no longer in the configuration.toml.
https://github.com/edgexfoundry/device-virtual-go/tree/v2.3.0/cmd/res/devices

Simply edit the devices definition file to add your device.

Note that EdgeX 1.x releases are no longer supported. Currently releases 2.1 (LTS) and 2.3 (STS) are supported.

**Reply by HCL-BMC** [reply]
Thanks for the response.

Now,  I can able to add the devices in the devices.toml.

But, I wanted to set the mode(Advertiser mode) for the devices which I have added.  For example, If I will add BLE device, How to set the Advertiser mode to that device. So that, it will act as advertiser and will start to advertising to make connection with other devices.

Could you please help for me for this and clarify me on this.

**Reply by iain-anderson** [reply]
Hi, the README for the device service implies that in the current implementation the device needs to be set to advertiser mode by other means before the service can connect to it.

**Reply by HCL-BMC** [reply]
> Hi, the README for the device service implies that in the current implementation the device needs to be set to advertiser mode by other means before the service can connect to it.

Yes. README file has the above details. But README file doesn't have the info like how to set the advertiser mode. Any reference config for that ? It will be helpful

Also, As my understanding current implementation and source code is not available for advertiser mode device. Pls clarify me on this.

Thanks for your response Iain.
